lib/linter-plus.coffee

```diff
@@ -31,12 +31,17 @@ class Linter
       @commands.lint()
 
     @subscriptions.add atom.workspace.observeTextEditors (editor) =>
-      currentEditorLinter = new EditorLinter @, editor
-      @editorLinters.set editor, currentEditorLinter
-      @emitter.emit 'observe-editor-linters', currentEditorLinter
-      currentEditorLinter.lint false
-      editor.onDidDestroy =>
-        currentEditorLinter.destroy()
+      editorLinter = new EditorLinter(editor)
+      @editorLinters.set editor, editorLinter
+      @emitter.emit 'observe-editor-linters', editorLinter
+      editorLinter.onShouldUpdateBubble =>
+        @views.renderBubble()
+      editorLinter.onShouldUpdateLineMessages =>
+        @views.updateLineMessages(true)
+      editorLinter.onShouldLint =>
```

Comment:
Why isn't this just contained inside `editorLinter`? Why is `EditorLinter` even emitting an event instead of just calling `@lint` internally?